phoenix-core/src/main/java/org/apache/phoenix/expression/function/RowTimestampEpochFunction.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.phoenix.expression.function;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode.BuiltInFunction;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PDate;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Function to return the timestamp of the empty column which functions as the row timestamp. The
+ * result returned can be used for debugging(eg. using HBase shell), logging etc.
+ */
+@BuiltInFunction(name = RowTimestampEpochFunction.NAME, args = {})
+public class RowTimestampEpochFunction extends ScalarFunction {
+
+    public static final String NAME = "ROW_TIMESTAMP_EPOCH";
+
+    public RowTimestampEpochFunction() {
+    }
+
+    public RowTimestampEpochFunction(List<Expression> children) { super(children); }
+
+    @Override
+    public String getName() { return NAME; }
+
+    @Override
+    public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
+        assert (tuple.size() >= 1);
```

Comment:
Do we use asserts in other functions? Agree there needs to be some validation logic here, just unsure of what the right exception is.

Reply:
I didn't find any asserts being used in other functions. Also didn't find any functions which explicitly raise exceptions. Intuitively, an assert made most sense here because you will always expect a non empty result tuple and that's what you are asserting on.